frontend/src/concepts/pipelines/content/__tests__/utils.spec.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable camelcase*/
+import {
+  BanIcon,
+  CheckCircleIcon,
+  ExclamationCircleIcon,
+  NotStartedIcon,
+  QuestionCircleIcon,
+  SyncAltIcon,
+} from '@patternfly/react-icons';
+import React from 'react';
+import {
+  PipelineRunKF,
+  PipelineRunStatusesKF,
+  RunStorageStateKF,
+} from '~/concepts/pipelines/kfTypes';
+import { computeRunStatus } from '~/concepts/pipelines/content/utils';
+
+const run: PipelineRunKF = {
+  created_at: '2023-09-05T16:23:25Z',
+  storage_state: RunStorageStateKF.AVAILABLE,
+  pipeline_spec: {
+    runtime_config: {
+      parameters: {},
+      pipeline_root: '',
+    },
+  },
+  finished_at: '2023-09-05T16:24:34Z',
+  id: 'dc66a214-4df2-4d4d-a302-0d02d8bba0e7',
+  name: 'flip-coin',
+  scheduled_at: '1970-01-01T00:00:00Z',
+  service_account: 'pipeline-runner-pipelines-definition',
+  status: '',
+  error: '',
+  metrics: [],
+};
+
+describe('computeRunStatus', () => {
+  const statuses = [undefined, 'Started', 'Running', 'Completed', 'Failed', 'Cancelled', 'warning'];
+
+  beforeEach(() => {
+    const status = statuses.shift();
+    run.status = status !== undefined ? status : 'undefined';
+  });
+
+  it('should check for run status when run status is undefined', () => {
+    const UNKNOWN_ICON = <QuestionCircleIcon />;
+    const UNKNOWN_STATUS = 'warning';
+    const UNKNOWN_LABEL = '-';
+    const runStatus = computeRunStatus();
+    expect(runStatus.label).toBe(UNKNOWN_LABEL);
+    expect(runStatus.icon).toStrictEqual(UNKNOWN_ICON);
+    expect(runStatus.status).toBe(UNKNOWN_STATUS);
+  });
+
+  it('should check for Started run status', () => {
+    const runStatus = computeRunStatus(run);
```

Comment:
This test file is fragile as each test is now tied to the array of `statuses` and the order of its contents.
Best to create a function for creating tailored pipeline runs per test.

```
const createRun = (status: string): PipelineRunKF => ({
  ...
  status,
  ...
});
```

Create a new run object per test:

```suggestion
    const runStatus = computeRunStatus(createRun('Started'));
```